spec/services/hwf_reference_generator_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe HwfReferenceGenerator, type: :service do
```

Comment:
And it'd be really good cause later we can generalise this to be used for our other reference generator and then somehow merge to be the same or very close to be the same.